### ISSUE TITLE
bug: ibis.struct doesn't work with `sqlglot<=20.1.0`

### ISSUE THREAD

**Issue by NickCrews** [ibis-project/ibis#8360]
### What happened?

```python
import ibis

t = ibis.examples.penguins.fetch()
t2 = t.select(s=ibis.struct({"a": t.island}))
ibis.to_sql(t2)

# python -m pip install "sqlglot==20.1.0"
# SELECT
#   {'island': } AS "s"
# FROM "penguins" AS "t0"

# python -m pip install "sqlglot==20.2.0"
# SELECT
#   {'a': "t0"."island"} AS "s"
# FROM "penguins" AS "t0"
```

I think this means we need to make the dependency restriction on sqlglot more strict?

### What version of ibis are you using?

main

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by gforsyth** [comment]
Thanks for reporting! Yeah, we will bump the minimum required version before we release 9.0.
We can leave this open as a tracking issue for that.

**Comment by cpcloud** [comment]
In the meantime I'll look into whether there's a way we can address this that is backwards compatible.

**Comment by NickCrews** [comment]
Yeah, I found this because I was given an old version of sqlglot by PDMs dependency solver, because I have some other dependency that has an upper bound on sqlglot of <20. I had to go and make a PR there so that sqlglot>=20 worked with their package.

This is an example of what happens when 2 libs support only a few versions of a shared dep, and it can become difficult to find a version that works for both. So it would be great if ibis could be a good citizen and try to support versions of sqlglot up to say 6 months old. I recognize that can be hard and sometimes just impossible, but just so it's on your radar.
